components/src/dynamo/frontend/tests/test_vllm_processor_unit.py

```
@@ -1104,3 +1104,100 @@ def test_runtime_config_context_length(vllm_processor_module, runtime_config, ex
+    The Rust OpenAIPreprocessor is bypassed on the vllm chat-processor path, so
+    this file owns the translation the Rust ``get_guided_json`` normally does.
+    The worker feeds the result into vLLM's ``StructuredOutputsParams``, which
+    rejects a constraint set holding zero or more than one constraint — so the
+    key must be absent unless exactly one constraint is present.
+    """
+
+    SCHEMA = {
+        "type": "object",
+        "properties": {"city": {"type": "string"}},
+        "required": ["city"],
+    }
```

Ruff (0.16.1)

[warning] Mutable default value for class attribute

(RUF012)